### DISCUSSION TITLE
TypeError: undefined is not an object (evaluating 'ReactSharedInternals.ReactCurrentOwner') with Three.js Library

### DISCUSSION THREAD

**Discussion by nimazabi** [vercel/next.js#72321] [Help] [answered]
### Summary

I've encountered an issue while using the [Three.js ] library in my project. The error message I'm receiving is: TypeError: undefined is not an object (evaluating 'ReactSharedInternals.ReactCurrentOwner'). I would appreciate any insights or solutions from the community to resolve this problem.
the predefined next js three js examples on internet works well, at the same time I updated packages it collapsed. is not reliable way to do my project!
<img width="969" alt="Screenshot 1403-08-15 at 10 57 05" src="https://github.com/user-attachments/assets/90e7541c-939e-47d4-922b-1663ee217d55">

### Additional information

```js
{
  "name": "spaceyv",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "@react-three/drei": "^9.115.0",
    "@react-three/fiber": "^8.17.10",
    "@types/tailwindcss": "^3.0.11",
    "@types/three": "^0.169.0",
    "clsx": "^2.1.1",
    "framer-motion": "^11.11.11",
    "framer-motion-3d": "^11.11.11",
    "lottie-react": "^2.4.0",
    "next": "^15.0.2",
    "react": "^18.3.1",
    "react-dom": "^18.3.1",
    "react-icons": "^5.3.0",
    "react-lottie-player": "^2.1.0",
    "swiper": "^11.1.14",
    "three": "^0.169.0"
  },
  "devDependencies": {
    "@svgr/webpack": "^8.1.0",
    "@types/node": "^20",
    "@types/react": "^18",
    "@types/react-dom": "^18",
    "eslint": "^8",
    "eslint-config-next": "15.0.1",
    "postcss": "^8",
    "tailwindcss": "^3.4.1",
    "typescript": "^5"
  }
}
```

**Comment by icyJoseph** [comment, marked as answer]
This is a problem with React 19 compatibility, according to: https://github.com/vercel/next.js/issues/66468#issuecomment-2296319798

Some claim that the alpha channel of react-three-fiber has a fix for this.

**Reply by nimazabi** [reply]
It actually solved when we updated to fiber alpha version. It means a lot to me at this moment.
thanks you

**Comment by edinsoncs** [comment]
The same thing happens
